projects/manage-supervision-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/sentence/entity/UpwAppointment.kt

```diff
@@ -0,0 +1,45 @@
+package uk.gov.justice.digital.hmpps.integrations.delius.sentence.entity
+
+import jakarta.persistence.Column
+import jakarta.persistence.Entity
+import jakarta.persistence.Id
+import org.hibernate.annotations.Immutable
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Query
+
+@Entity
+@Immutable
+class UpwAppointment(
+    @Id
+    @Column(name = "upw_appointment_id")
+    val id: Long,
+
+    val minutesCredited: Long?,
+
+    @Column(columnDefinition = "char(1)")
+    val attended: String?,
+
+    val softDeleted: Long,
+
+    val offenderId: Long,
+)
+
+interface UpwAppointmentRepository : JpaRepository<UpwAppointment, Long> {
+
+    @Query(
+        """
+            SELECT nvl(sum(duration), 0) FROM (
+                SELECT ua.MINUTES_CREDITED AS duration
+                FROM UPW_APPOINTMENT ua
+                JOIN OFFENDER o 
+                ON o.OFFENDER_ID = ua.OFFENDER_ID 
+                JOIN EVENT e 
+                ON e.OFFENDER_ID = o.OFFENDER_ID 
+                WHERE e.EVENT_ID = :id
+                AND e.EVENT_NUMBER = :eventNumber
+                AND ua.ATTENDED = 'Y'
+                AND ua.SOFT_DELETED = 0)
```

Comment:
I'm not sure you need this sub-select.  Also, this looks like it'll get any UPW appointments for the offender, even if the appointments are for other events?  You might be able to link to the sentence (disposal) via the upw_details, something like:

```suggestion
            select sum(minutes_credited) 
            from upw_appointment 
            join upw_details on upw_details.upw_details_id = upw_appointment.upw_details_id
            join disposal on disposal.disposal_id = upw_details.disposal_id
            where disposal.event_id = :eventId
            and upw_appointment.attended = 'Y'
            and upw_appointment.soft_deleted = 0
```

However at that point, are we sure we should be returning this as part of the Requirement model - if it's actually linked to the Sentence?

Reply:
sql updated.  i am happy to leave the model at the moment